### DISCUSSION TITLE
Access RSID from config downloaded using

### DISCUSSION THREAD

**Discussion by thomas-chris** [adobe/aepsdk-core-ios#821] [General]
I've registered my app using `MobileCore.configureWith(appId: adobeAppId)` in the completion of `MobileCore.registerExtensions`.

If i've understood what this is doing it replaces the need to have a file manually added to my bundle (which is how things used to be set up when I was on v4 stuff that is deprecated.

I thought looking I should be able to use `MobileCore.getSdkIdentities` to fetch the values from this file, but that doesn't appear so. I want to get access to the RSID value that is sent in the json that is downloaded at app launch, how can i do this?

**Comment by cdhoffmann** [comment]
@thomas-chris Hello. I apologize for the lack of documentation on the subject, we are working on migrating our docs and will have better documentation on this subject soon. There is a bit to unwrap here so let me try and explain. 

The file you are talking about manually adding to the bundle, which includes the RSID value, is called `Configuration`. The configuration file is downloaded remotely upon registering your extensions. [MobileCore.getSdkIdentities](https://aep-sdks.gitbook.io/docs/foundation-extensions/mobile-core/mobile-core-api-reference#getsdkidentities) should return the `analytics.rsids` value as part of the completion handler's data as long as the [user is opted-in](https://aep-sdks.gitbook.io/docs/resources/privacy-and-gdpr#set-and-get-privacy-status). So, under these conditions, you might not be getting the RSID value when you call `MobileCore.getSdkIdentities` if:

1. The remote configuration has not yet been downloaded before you call it. (if you are calling `MobileCore.getSdkIdentities` immediately after registration for instance, you are probably not giving enough time for the configuration to download.
2. The privacy status is set to `optedOut`

There are a couple of things you can do to work around these conditions:
1. Use a [bundled Configuration](https://aep-sdks.gitbook.io/docs/foundation-extensions/mobile-core/configuration#using-a-bundled-file-configuration) to ensure that calling `MobileCore.getSdkIdentities` returns the RSID value **as long as the user is opted in**. 
2. Use the `MobileCore.registerEventListener` API to listen for a `configurationResponse` event. This will trigger a callback when the remote configuration has been downloaded with the configuration. Here is an example of this API in action for your use case:

```
        MobileCore.registerEventListener(type: "Configuration Response Event", source: "com.adobe.eventSource.responseContent") { event in

            if let rsid = event.data?["analytics.rsids"] {
                
            }

        }
```

Please let me know if this works for you or if you have any other questions. Thanks.

Chris
